src/components/Carousel/components/Slide/Slide.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2019, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import { css, keyframes } from '@emotion/core';
+
+import { childrenPropType } from '../../../../util/shared-prop-types';
+
+import * as SlideService from './SlideService';
+import { ANIMATION_DURATION, SLIDE_DIRECTIONS } from '../../constants';
+
+const baseStyles = ({ index, stackOrder, width }) => css`
+  label: carousel__slide;
+  width: 100%;
+  flex-grow: 0;
+  flex-shrink: 0;
+  flex-basis: ${width};
+  transform: translate3d(${-index * 100}%, 0, 0);
+  backface-visibility: hidden;
+  position: relative;
+  z-index: ${stackOrder};
+`;
+const Wrapper = styled('div')(baseStyles);
+
+const slideIn = keyframes`
+  from {
+    width: 0%;
```

Comment:
Is it possible to use `transform: scaleX` instead of `width`? This way the animation can be optimised by the GPU.

Reply:
it's possible but animation doesn't look the way it should be unfortunately.

![carousel-transform-scalex](https://user-images.githubusercontent.com/974035/73083032-bfc69580-3eca-11ea-8065-12a0bd5352e5.gif)